outlines/caching.py

```diff
@@ -17,6 +20,28 @@ def cache_fn(fn: Callable):
     return cache_fn
 
 
+def hash_it(*data) -> str:
```

Comment:
This is directly taken from perscache library. The difference in cache size b/w a hashed key and just tupled key is quite a lot: `224K` and `1.1M` respectively.

Reply:
```

~/outlines$ ls -lh ~/.cache/outlines/
total 212K
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 00
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 02
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 07
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 0c
drwxr-xr-x 4 avirlrma avirlrma 4.0K Dec 17 07:54 0f
drwxr-xr-x 4 avirlrma avirlrma 4.0K Dec 17 07:54 13
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 18
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 1a
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 25
drwxr-xr-x 4 avirlrma avirlrma 4.0K Dec 17 07:54 26
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 2e
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 3c
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 41
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 42
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 43
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 46
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 49
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 4b
drwxr-xr-x 4 avirlrma avirlrma 4.0K Dec 17 07:54 51
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 56
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 65
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 6a
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 73
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 82
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 88
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 92
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 94
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 a5
drwxr-xr-x 4 avirlrma avirlrma 4.0K Dec 17 07:54 ab
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 ae
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 b5
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 b7
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 ba
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 c3
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 c4
-rw-r--r-- 1 avirlrma avirlrma  36K Dec 17 07:54 cache.db
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 cc
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 cd
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 d7
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 e0
drwxr-xr-x 4 avirlrma avirlrma 4.0K Dec 17 07:54 e1
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 ec
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 ee
drwxr-xr-x 4 avirlrma avirlrma 4.0K Dec 17 07:54 f2
drwxr-xr-x 3 avirlrma avirlrma 4.0K Dec 17 07:54 f8

```